AdyenDropIn/Views/ModalToolbar.swift

```diff
@@ -0,0 +1,163 @@
+        stack.spacing = 16
+        stack.layoutMargins = .init(top: 0, left: paddingWithMarginCorrection, bottom: 0, right: paddingWithMarginCorrection)
+        stack.isLayoutMarginsRelativeArrangement = true
+
+        return stack
+    }()
+
+    @objc private func didCancel() {
+        cancelHandler()
+    }
+
+    // MARK: - View setup
+
+    private func setupView() {
+        switch style.barTitle.textAlignment {
+        case .center:
+            addSubview(titleLabel)
+            addSubview(cancelButton)
+            setupLayoutForCenteredMode()
+        default:
+            addSubview(stackView)
+            anchore(view: self, toView: stackView)
+        }
+
+        setupStyle()
+    }
+
+    private func setupStyle() {
+        self.backgroundColor = style.backgroundColor
+        cancelButton.tintColor = style.tintColor
+        guard let title = titleLabel.text, !title.isEmpty else { return }
+        titleLabel.attributedText = NSAttributedString(string: title,
+                                                       attributes: [NSAttributedString.Key.font: style.barTitle.font,
+                                                                    NSAttributedString.Key.foregroundColor: style.barTitle.color,
+                                                                    NSAttributedString.Key.backgroundColor: style.barTitle.backgroundColor])
+    }
+
+    private func setupLayoutForCenteredMode() {
+        let rightAnchor: NSLayoutXAxisAnchor
+        let leftAnchor: NSLayoutXAxisAnchor
+        if #available(iOS 11.0, *) {
+            rightAnchor = self.safeAreaLayoutGuide.rightAnchor
+            leftAnchor = self.safeAreaLayoutGuide.leftAnchor
+        } else {
+            rightAnchor = self.rightAnchor
+            leftAnchor = self.leftAnchor
+        }
+
+        let cancePositionConstraint: NSLayoutConstraint
+        if style.toolbarMode == .rightCancel {
+            cancePositionConstraint = rightAnchor.constraint(equalTo: cancelButton.rightAnchor, constant: paddingWithMarginCorrection)
+        } else {
+            cancePositionConstraint = leftAnchor.constraint(equalTo: cancelButton.leftAnchor, constant: -paddingWithMarginCorrection)
+        }
+
+        NSLayoutConstraint.activate([
+            self.topAnchor.constraint(equalTo: titleLabel.topAnchor),
+            self.bottomAnchor.constraint(equalTo: titleLabel.bottomAnchor),
+            leftAnchor.constraint(equalTo: titleLabel.leftAnchor, constant: -paddingWithMarginCorrection),
+            rightAnchor.constraint(equalTo: titleLabel.rightAnchor, constant: paddingWithMarginCorrection),
+            
+            self.centerYAnchor.constraint(equalTo: cancelButton.centerYAnchor),
+            self.heightAnchor.constraint(greaterThanOrEqualTo: cancelButton.heightAnchor),
+            cancePositionConstraint
+        ])
+    }
+
+    // MARK: - Private methods
+
+    private func anchore(view view1: UIView, toView view2: UIView) {
```

Comment:
this can be a UIView extension helper, inside the AdyenScope.

Reply:
yep.. why not :D